lib/authentication_failure_app.rb

```diff
@@ -0,0 +1,47 @@
+class AuthenticationFailureApp < Devise::FailureApp
+  # This code is 99% reproduced from the Devise gem's Devise::FailureApp class
+  # at: lib/devise/failure_app.rb. The only exception is that we comment out the 
+  # 5th line:
+  #   opts[:script_name] = nil
+  #
+  # This line was causing the`env["SCRIPT_NAME"]` value that's set by
+  # OrganizationMiddleware to not be used when generating the route in the
+  # statement `context.send(route, opts)`, which wipes out the current
+  # Organization context on authentication failure.
```

Comment:
Ah very nice! Great find. I have not worked with this module before so I learned something new. 

At the risk of writing a novel, could we add just a bit more context in these comments to provide a crystal clear explanation of why this is important? I think what you have is good, but you could just add something like 'I.e., if an unauthenticated user tried to access an authenticated route within an organization's scope, they would be redirected to the sign in page, but the organization scope would be lost in the URL. This fixes that to make sure the Organization's slug is present in the URL' or something like that?

Hopefully this was an interesting journey for you, too! I learned something.

Reply:
Will do! Yeah, I enjoyed this issue and learned some new concepts as well. I've used Devise a few times on some personal projects but never with one this large! I'm sure there are other ways to achieve the same end result, but I quite like this method.

I noticed there are some merge conflicts; should I take care of those too?

Reply:
I added some more detail on what this module/method is doing. Let me know if it sounds clear enough or if it needs to be trimmed down!

Reply:
Yes that looks good thank you!